### PR TITLE
Add op(unfold) | feat(torchlib)

### PR DESCRIPTION
**Pull request by xiaowuhu**
from: https://github.com/microsoft/onnxscript/pull/534

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -6725,10 +6725,46 @@ def aten_unflatten(self: TReal, dim: INT64, sizes: INT64):
...
+    if self_rank == 0:
+        result = op.Unsqueeze(self, 0)
+    else:
+        dim_size = self.shape[dimension]
+        target_end = (dim_size - size) // step + 1
+        if target_end > 1:  # the rank of final reuslt will be self_rank + 1
+            self_rank = self_rank + 1
+        # perm need to be list[int], so have to be generated in trace_only mode
+        perm = list(range(self_rank))
+        perm.append(perm.pop(dimension + 1))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:6742]: Would be great to add a comment to explain this logic, thanks!

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 2]
```diff
@@ -1164,6 +1164,11 @@ def _where_input_wrangler(
         matcher=lambda sample: any(dim == 0 for dim in sample.input.shape),
         reason="fixme: Logic not implemented for size 0 inputs in op.Reshape",
     ),
+    TorchLibOpInfo(
+        "unfold",
+        core_ops.aten_unfold,
+        trace_only=True,
+    ),
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:1171]:
```suggestion
    TorchLibOpInfo("unfold", core_ops.aten_unfold, trace_only=True),
```

nit: can be more compact

**Review by justinchuby — COMMENTED** [review]
LGTM. Would be good to get another review from @BowenBao

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (xiaowuhu) [reply onnxscript/function_libs/torch_lib/ops/core.py:6742]: add comment.

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py (hunk 2 above)
Comment (xiaowuhu) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py]: changed

**Review by justinchuby — APPROVED** [review]

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:6730]: Why do we do this though? What does it mean? Is there a reference logic?

Comment (xiaowuhu) [reply onnxscript/function_libs/torch_lib/ops/core.py:6730]: 1.  similar but different in detail: https://github.com/pytorch/pytorch/blob/a8f40b39ce4f9fa9ffd90400b7d10ea4051d623a/torch/onnx/symbolic_opset12.py#L390
2. the same logic: https://github.com/microsoft/onnxscript/blob/50c3ee968960f9f7735df3f0f2458643d9d9ba08/onnxscript/function_libs/torch_lib/ops/core.py#L5999

**Review by BowenBao — APPROVED** [review]

**Pull request merged**